examples/basics/demo_dr_methods_affinity_matcher.py

```diff
@@ -0,0 +1,111 @@
+r"""
+Comparison of different DR methods and the use of affinity matcher
+========================================
```

Comment:
the `====` line should be the same length as the line above, otherwise sphinx usually complains